demos/metrics_demo.cc

```diff
@@ -0,0 +1,137 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <random>
+#include <chrono>
+#include <seastar/core/app-template.hh>
+#include <seastar/core/seastar.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/util/log.hh>
+#include <seastar/http/httpd.hh>
+#include <seastar/core/thread.hh>
+#include <seastar/core/prometheus.hh>
+#include <seastar/util/defer.hh>
+#include <seastar/core/metrics.hh>
+#include <seastar/core/metrics_api.hh>
+#include <seastar/core/metrics_registration.hh>
+#include <fmt/format.h>
+
+using namespace seastar;
+using namespace std::chrono_literals;
+namespace sm = seastar::metrics;
+namespace bpo = boost::program_options;
+
+logger applog("app");
+
+class metrics_example {
+private:
+    sm::metric_groups _metrics;
+    uint64_t _counter;
+    uint64_t _triggered;
+
+public:
+    metrics_example(int instance) : _counter(0) {
+        // Note that these metrics are only available as long at the _metrics instance
+        // is valid.
+        _metrics.add_group(fmt::format("metrics_demo_{}", instance), {
```

Comment:
If it was a real code, I would say instane should be a label, I know it's a demo, but maybe we should
be correct.